### PR TITLE
feat: smart account signer

### PR DESCRIPTION
**Pull request by AmanRaj1608**

### CODE REVIEW

File: packages/smart-account/src/signers/Signer.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { BytesLike } from '@ethersproject/bytes'
 import { Deferrable } from 'ethers/lib/utils'
 import { TransactionRequest } from '@ethersproject/providers'
 
-export abstract class Signer extends AbstractSigner {
+export abstract class Signer extends EthersSigner {
```
Comment (livingrockrises) [inline packages/smart-account/src/signers/Signer.ts:14]: Are we using this Signer class directly? 
Why are we extending from Signer and not AbstractSigner?

File: packages/smart-account/src/signers/SmartAccountSigner.ts [hunk 2]
```diff
@@ -1,47 +1,34 @@
...
 
   _address!: string
+  // relayer: Relayer
 
   // Might have
   // _context: not smartAccountContext but the addresses of contracts from SmartAccountState
-  //
 
   // TBD
-  private _providers: { [key: number]: Web3Provider } = {}
```
Comment (livingrockrises) [inline packages/smart-account/src/signers/SmartAccountSigner.ts:44]: Why is this needed?

File: packages/smart-account/src/signers/SmartAccountSigner.ts [hunk 3]
```diff
@@ -53,25 +40,28 @@ export class SmartAccountSigner extends Signer implements TypedDataSigner {
...
+  //   console.log(chainId)
+  //   throw new Error('TODO')
+  // }
 
+  // Review getProvider
+  // getProvider returns a JsonRpcProvider instance for the current chain.
+  // Note that this method is bound to a particular chain
   // Review for the provider we want here
-  async getProvider(chainId?: number): Promise<Web3Provider | undefined> {
+  async getProvider(chainId?: number): Promise<JsonRpcProvider | undefined> {
```
Comment (livingrockrises) [inline packages/smart-account/src/signers/SmartAccountSigner.ts:64]: for example are this methods accessed anyway in any other classes where this will be passed as original signer?

**Comment by livingrockrises** [comment]
let me know when it's ready to merge

File: packages/smart-account/src/signers/Signer.ts (hunk 1 above)
Comment (AmanRaj1608) [reply packages/smart-account/src/signers/Signer.ts]: Updated!

File: packages/smart-account/src/signers/SmartAccountSigner.ts (hunk 2 above)
Comment (AmanRaj1608) [reply packages/smart-account/src/signers/SmartAccountSigner.ts:44]: It was there in the code when I started working on this, had removed it.
